ocaml/xapi/repository.ml

```diff
@@ -608,27 +608,31 @@ let get_pool_updates_in_json ~__context ~hosts =
         Hashtbl.fold
           (fun host updates_of_host (acc1, acc2) ->
             if List.mem host hosts then
-              let json_of_host, upd_ids =
+              let updates_of_host, upd_ids =
                 consolidate_updates_of_host ~repository_name ~updates_info
                   (Ref.string_of host) updates_of_host
               in
-              (json_of_host :: acc1, UpdateIdSet.union upd_ids acc2)
+              (updates_of_host :: acc1, UpdateIdSet.union upd_ids acc2)
             else
               (acc1, acc2)
           )
           updates_in_cache ([], UpdateIdSet.empty)
       in
+      let lps =
+        updates_of_hosts
+        |> List.map (fun x -> x.HostUpdates.livepatches)
+        |> List.concat
+        |> LivePatchSet.of_list
+      in
+      let updateinfo_list =
+        UpdateIdSet.elements ids_of_updates
+        |> List.map (fun upd_id -> List.assoc upd_id updates_info)
```

Comment:
Perhaps we can change updates_info to be a map in the future? (e.g. separate PR)

Reply:
Yeah. This makes sense. I created an improvement for this. Thanks.